room/runtime/src/main/java/androidx/room/InvalidationTracker.java

```diff
@@ -389,7 +399,7 @@ public void run() {
                     // This transaction has to be on the underlying DB rather than the RoomDatabase
                     // in order to avoid a recursive loop after endTransaction.
                     SupportSQLiteDatabase db = mDatabase.getOpenHelper().getWritableDatabase();
-                    db.beginTransaction();
+                    db.beginTransactionNonExclusive();
```

Comment:
This line should call the new internal method, `beginTransactionInternal`

Reply:
This is line is already inside `if (mDatabase.mWriteAheadLoggingEnabled)`, so I thought it's better to to use the native method?

Reply:
Oh right, I missed that! Thanks!